core/camel-api/src/main/java/org/apache/camel/Exchange.java

```diff
@@ -254,6 +254,7 @@ public interface Exchange {
     String TRACING_HEADER_FORMAT = "CamelTracingHeaderFormat";
     @Deprecated
     String TRACING_OUTPUT_FORMAT = "CamelTracingOutputFormat";
+    String TRANSACTION_CONTEXT_DATA = "CamelTransactionContextData";
```

Comment:
its a bit generic this name? Do you need to store something else than the EM factory?

TRANSACTION_MANAGER

or 

TRANSACTION_ENTITY_MANAGER

Reply:
Currently it only needs to store an EntityManager but I'd like to keept it more generic since it could be used for sharing any data in sub exchanges during the transaction.